hare3/hare_test.go

```diff
@@ -699,6 +699,61 @@ func TestHandler(t *testing.T) {
 	})
 }
 
+func TestHandlerOwnEquivocation(t *testing.T) {
+	t.Parallel()
+	tst := &tester{
+		TB:            t,
+		rng:           rand.New(rand.NewSource(1001)),
+		start:         time.Now(),
+		cfg:           DefaultConfig(),
+		layerDuration: 5 * time.Minute,
+		beacon:        types.Beacon{1, 1, 1, 1},
+		genesis:       types.GetEffectiveGenesis(),
+	}
+	cluster := newLockstepCluster(tst)
+	cluster.addActive(1)
+	n := cluster.nodes[0]
+	require.NoError(t, beacons.Add(n.db, tst.genesis.GetEpoch()+1, tst.beacon))
+	require.NoError(t, atxs.Add(n.db, n.atx))
+	n.oracle.UpdateActiveSet(tst.genesis.GetEpoch()+1, []types.ATXID{n.atx.ID()})
+	n.mpublisher.EXPECT().Publish(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	layer := tst.genesis + 1
+	n.nclock.StartLayer(layer)
+	n.clock.Set(tst.start.
+		Add(tst.layerDuration * time.Duration(layer)).
+		Add(tst.cfg.PreroundDelay))
+	elig := n.tracer.waitEligibility()
+
+	msg1 := &Message{}
+	msg1.Layer = layer
+	msg1.Value.Proposals = []types.ProposalID{{1}}
+	msg1.Eligibility = *elig
+	msg1.Sender = n.signer.NodeID()
+	msg1.Signature = n.signer.Sign(signing.HARE, msg1.ToMetadata().ToBytes())
+
+	msg2 := &Message{}
+	msg2.Layer = layer
+	msg2.Value.Proposals = []types.ProposalID{{2}}
+	msg2.Eligibility = *elig
+	msg2.Sender = n.signer.NodeID()
+	msg2.Signature = n.signer.Sign(signing.HARE, msg2.ToMetadata().ToBytes())
+
+	types.SetMinerNodeID(n.hare.signer.NodeID())
```

Comment:
this global state can't work with t.Parallel, but hare already has access to signer